app/scripts/directives/fromFile.js

```diff
@@ -157,6 +157,10 @@ angular.module("openshiftConsole")
               openTemplateProcessModal();
             // Else if any resources already exist
             } else if (!_.isEmpty($scope.updateResources)) {
+              // If the updated resource is a Template, set the process option to false
+              if ($scope.updateResources.length === 1 && $scope.updateResources[0].kind === "Template") {
+                $scope.templateOptions.process = false;
```

Comment:
I think the best experience would be to show the same dialog, but ask if they want to update the template or process it (instead of create and process). At least if there aren't other items in the YAML. @jhadvig how hard would that be?